lib/eventHubReceiver.ts

```diff
@@ -372,20 +306,177 @@ export class EventHubReceiver extends EventEmitter {
     }
   }
 
+  /**
+   * Starts the receiver by establishing an AMQP session and an AMQP receiver link on the session.
+   * This method **MUST NOT** be used along with the `receive()` method. Once you start the receiver
+   * the only way to stop the receiver is by closing the receiver or an error occurs while receiving
+   * the messages.
+   *
+   * @param {OnMessage} onMessage The message handler to receive event data objects.
+   * @param {OnError} onError The error handler to receive an error that occurs while receivin messages.
+   */
+  start(onMessage: OnMessage, onError: OnError): void {
```

Comment:
Would `receiveWithHandlers()` be a better name compared to `start()`. That will automatically put a lid on the confusion about how to stop receiving messages once you have started something.
There is a close() on the receiver, sender and the client that closes the underlying amqp links.